src/compiler/checker.ts

```diff
@@ -17535,11 +17535,10 @@ namespace ts {
         }
 
         function checkSourceFile(node: SourceFile) {
-            const start = performance.mark();
-
+            performance.mark("checkStart");
             checkSourceFileWorker(node);
-
-            performance.measure("Check", start);
+            performance.mark("checkEnd");

```

Comment:
I call `mark` to emit the performance event, that way I can query a cpuprofile for things like `between(checkStart, checkEnd)`.